crates/sargon/src/types/hd_signature.rs

```diff
@@ -0,0 +1,114 @@
+use crate::prelude::*;
+
+/// A signature of `intent_hash` by `entity` using `factor_source_id` and `derivation_path`, with `public_key` used for verification.
+#[derive(Clone, PartialEq, Eq, Hash, derive_more::Debug)]
+#[debug("HDSignature {{ input: {:#?} }}", input)]
+pub struct HDSignature {
+    /// The input used to produce this `HDSignature`
+    pub input: HDSignatureInput,
+
+    /// The ECDSA/EdDSA signature produced by the private key of the
+    /// `owned_hd_factor_instance.public_key`,
+    /// derived by the HDFactorSource identified by
+    /// `owned_hd_factor_
+    /// instance.factor_s
+    /// ource_id` and which
+    /// was derived at `owned_hd_factor_instance.derivation_path`.
+    pub signature: Signature,
+}
+
+impl HDSignature {
+    /// Constructs a HDSignature from an already produced `Signature`.
+    pub(crate) fn with_details(
+        input: HDSignatureInput,
+        signature: Signature,
+    ) -> Self {
+        Self { input, signature }
+    }
+
+    pub fn intent_hash(&self) -> &IntentHash {
+        &self.input.intent_hash
+    }
+
+    pub fn owned_factor_instance(&self) -> &OwnedFactorInstance {
+        &self.input.owned_factor_instance
+    }
+
+    pub fn factor_source_id(&self) -> FactorSourceIDFromHash {
+        self.owned_factor_instance()
+            .factor_instance()
+            .factor_source_id
+    }
+
+    pub fn derivation_path(&self) -> DerivationPath {
+        self.input
+            .owned_factor_instance
+            .factor_instance()
+            .derivation_path()
+    }
+}
+
+impl HasSampleValues for HDSignature {
+    fn sample() -> Self {
+        Self::sample_from_input(HDSignatureInput::sample())
+    }
+
+    fn sample_other() -> Self {
+        Self::sample_from_input(HDSignatureInput::sample_other())
+    }
+}
+
+impl HDSignature {
+    // Signs with predefined mnemonics associated to the input's factor source id
+    pub fn sample_from_input(input: HDSignatureInput) -> Self {
```

Comment:
I really want this to be `cfg(test)` only. And the name is misleading, it does not produce sample signatures. maybe `sign_by_looking_up_mnemonic_amongst_samples`. 

My idea was that you put this directly in `TestSigningPolyInteractor` (`TestSigningParallelInteractor`) and then let `TestSigningMonoInteractor` (`TestSigningSerialInteractor`) use the `TestSigningPolyInteractor` (translating between the different shapes, essentially multiple values vs single values). 

Then we remove this `sample_from_input` and we remove `HDSignature:: produced_signing_with_input` - and any other simpler unit tests which rely on `HDSignature ::produced_signing_with_input` can then use `TestSigningMonoInteractor` directly!